internal/api/v3/handler_payments_create.go

```diff
@@ -0,0 +1,224 @@
+		return fmt.Errorf("invalid type: %w", err)
+	}
+
+	if r.Scheme == "" {
+		return errors.New("scheme is required")
+	}
+
+	if _, err := models.PaymentSchemeFromString(r.Scheme); err != nil {
+		return fmt.Errorf("invalid scheme: %w", err)
+	}
+
+	if r.Asset == "" {
+		return errors.New("asset is required")
+	}
+
+	if r.SourceAccountID != nil {
+		_, err := models.AccountIDFromString(*r.SourceAccountID)
+		if err != nil {
+			return fmt.Errorf("invalid sourceAccountID: %w", err)
+		}
+	}
+
+	if r.DestinationAccountID != nil {
+		_, err := models.AccountIDFromString(*r.DestinationAccountID)
+		if err != nil {
+			return fmt.Errorf("invalid destinationAccountID: %w", err)
+		}
+	}
+
+	if len(r.Adjustments) == 0 {
+		return errors.New("adjustments is required")
+	}
+
+	for i, adj := range r.Adjustments {
+		if err := adj.validate(); err != nil {
+			return fmt.Errorf("adjustment %d: %w", i, err)
+		}
+	}
+
+	return nil
+}
+
+func (r *createPaymentsAdjustmentsRequest) validate() error {
+	if r.Reference == "" {
+		return errors.New("reference is required")
+	}
+
+	if r.CreatedAt.IsZero() || r.CreatedAt.After(time.Now()) {
+		return errors.New("createdAt is empty or in the future")
+	}
+
+	if r.Amount == nil {
+		return errors.New("amount is required")
+	}
+
+	if r.Asset == nil {
+		return errors.New("asset is required")
+	}
+
+	if r.Status == "" {
+		return errors.New("status is required")
+	}
+
+	if _, err := models.PaymentStatusFromString(r.Status); err != nil {
+		return fmt.Errorf("invalid status: %w", err)
+	}
+
+	return nil
+}
+
+func paymentsCreate(backend backend.Backend) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx, span := otel.Tracer().Start(r.Context(), "v2_paymentsCreate")
+		defer span.End()
+
+		var req createPaymentRequest
+		err := json.NewDecoder(r.Body).Decode(&req)
+		if err != nil {
+			otel.RecordError(span, err)
+			api.BadRequest(w, ErrMissingOrInvalidBody, err)
+			return
+		}
+
+		if err := req.validate(); err != nil {
+			otel.RecordError(span, err)
+			api.BadRequest(w, ErrValidation, err)
+			return
+		}
+
+		connectorID := models.MustConnectorIDFromString(req.ConnectorID)
+		paymentType := models.MustPaymentTypeFromString(req.Type)
+		pid := models.PaymentID{
+			PaymentReference: models.PaymentReference{
+				Reference: req.Reference,
+				Type:      paymentType,
+			},
+			ConnectorID: connectorID,
+		}
+
+		payment := models.Payment{
+			ID:            pid,
+			ConnectorID:   connectorID,
+			Reference:     req.Reference,
+			CreatedAt:     req.CreatedAt.UTC(),
+			Type:          paymentType,
+			InitialAmount: req.InitialAmount,
+			Amount:        req.Amount,
+			Asset:         req.Asset,
+			Scheme:        models.MustPaymentSchemeFromString(req.Scheme),
+			SourceAccountID: func() *models.AccountID {
+				if req.SourceAccountID == nil {
+					return nil
+				}
+				return pointer.For(models.MustAccountIDFromString(*req.SourceAccountID))
```

Comment:
Same feedback here as in the v2 handler

Reply:
Already parsed also inside the validate function